lib/logstash/codecs/cef.rb

```diff
@@ -103,14 +103,31 @@ def decode(data)
         message=message + ' ' unless message.end_with?('\=')
       end
 
-      # Now parse the key value pairs into it
-      extensions = {}
-      message = message.split(/ ([\w\.]+)=/)
-      key, value = message.shift.split('=', 2)
-      extensions[key] = value.gsub(/\\=/, '=').gsub(/\\\\/, '\\')
-      Hash[*message].each{ |k, v| extensions[k] = v }
-      # And save the new has as the extensions
-      event.set('cef_ext', extensions)
+      # Introducing custom delimiters instead of space to avoid convolution when parsing fields with ' ' spaces.
+      # Example : cs1Label=New Malware cs1=Conficker ==> cs1Label=New Malware|^^^cs1=Conficker
+      message=message.gsub((/\s+(\w+=)/)||(/(\w+=)/),'|^^^\1')
```

Comment:
What is this supposed to do? you have `||` between to regular expressions. This operation, in ruby, seems to just result in the first regex being used:

```
>> /foo/ || /bar/
=> /foo/
```

I'm not sure what's supposed to be going on here, but this code is weird. Can you help me understand?